tests/infrastructure/golden_images/update_boot_source/test_boot_sources_vm.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+import pytest
+from ocp_resources.data_source import DataSource
+
+from utilities.infra import assert_os_version_mismatch_in_vm, validate_os_info_vmi_vs_linux_os
+from utilities.storage import data_volume_template_with_source_ref_dict
+from utilities.virt import VirtualMachineForTests, running_vm
+
+LOGGER = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def boot_source_preference_from_data_source_dict(data_import_cron_matrix__function__):
+    return data_import_cron_matrix__function__[[*data_import_cron_matrix__function__][0]]["preference"]
+
+
+@pytest.fixture()
+def data_source_from_data_import_cron(
+    golden_images_namespace,
+    golden_images_persistent_volume_claims_scope_function,
+    golden_images_volume_snapshot_scope_function,
+    data_import_cron_matrix__function__,
+):
+    data_source = DataSource(name=[*data_import_cron_matrix__function__][0], namespace=golden_images_namespace.name)
+    data_source.wait_for_condition(
+        condition=data_source.Condition.READY, status=data_source.Condition.Status.TRUE, timeout=5
+    )
+    return data_source
+
+
+@pytest.fixture()
+def auto_update_boot_source_instance_type_vm(
+    unprivileged_client,
+    namespace,
+    modern_cpu_for_migration,
+    data_source_from_data_import_cron,
+):
+    LOGGER.info(f"Create a VM using {data_source_from_data_import_cron.name} dataSource")
+    with VirtualMachineForTests(
+        client=unprivileged_client,
+        name=f"{data_source_from_data_import_cron.name}-data-source-vm",
+        namespace=namespace.name,
+        vm_instance_type_infer=True,
+        vm_preference_infer=True,
+        data_volume_template=data_volume_template_with_source_ref_dict(
+            data_source=data_source_from_data_import_cron,
+        ),
+        cpu_model=modern_cpu_for_migration,
+    ) as vm:
+        running_vm(vm=vm)
+        yield vm
+
+
+@pytest.mark.polarion("CNV-7586")
+def test_instance_type_vm_from_auto_update_boot_source(
+    latest_fedora_release_version,
+    auto_update_boot_source_instance_type_vm,
+    boot_source_preference_from_data_source_dict,
+):
+    LOGGER.info(f"Verify {auto_update_boot_source_instance_type_vm.name} OS version and virtctl info")
+    if "fedora" in auto_update_boot_source_instance_type_vm.name and latest_fedora_release_version:
+        boot_source_preference_from_data_source_dict = f"fedora{latest_fedora_release_version}"
```

Comment:
maybe a fixture?

Reply:
Ditto. Please don't resolve comments that you have not addressed.